src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -247,18 +246,16 @@ void DefaultDACVerifier::VerifyAttestationInformation(const DeviceAttestationVer
 
     // if PAA contains VID, see if matches with DAC's VID.
     {
-        uint16_t paaVid  = VendorId::NotSpecified;
-        CHIP_ERROR error = ExtractDNAttributeFromX509Cert(MatterOid::kVendorId, paaDerBuffer, paaVid);
-        VerifyOrExit(error == CHIP_NO_ERROR || error == CHIP_ERROR_KEY_NOT_FOUND,
+        VerifyOrExit(ExtractVIDPIDFromX509Cert(paaDerBuffer, paaVidPid) == CHIP_NO_ERROR,
                      attestationError = AttestationVerificationResult::kPaaFormatInvalid);
-        if (error != CHIP_ERROR_KEY_NOT_FOUND)
+        if (paaVidPid.mVendorId.HasValue())
         {
-            VerifyOrExit(paaVid == dacVendorId, attestationError = AttestationVerificationResult::kDacVendorIdMismatch);
+            VerifyOrExit(dacVidPid.mVendorId == paaVidPid.mVendorId,
+                         attestationError = AttestationVerificationResult::kDacVendorIdMismatch);
         }
     }
```

Comment:
Since PAI always requires VID, the step above that checks PAI VID could do check that if PAA has the VID, it matches the mandatory VID in PAI. Then, here, we could just check that DAC VID matches PAI PID, since the PAA chaining could have been done before.